code/modules/hydroponics/gene_modder.dm

```diff
@@ -480,19 +480,54 @@
 
 /obj/item/disk/plantgene/update_name()
 	. = ..()
+	if(HAS_TRAIT(src, TRAIT_CMAGGED))
+		name = "nuclear authentication disk"
+		return
 	if(gene)
 		name = "[gene.get_name()] (Plant Data Disk)"
 	else
 		name = "plant data disk"
 
+/obj/item/disk/plantgene/update_desc()
+	. = ..()
+	if(HAS_TRAIT(src, TRAIT_CMAGGED))
+		desc = "Better keep this safe."
+	else
+		desc = "A disk for storing plant genetic data."
+
+/obj/item/disk/plantgene/update_icon()
+	. = ..()
+	if(HAS_TRAIT(src, TRAIT_CMAGGED))
+		icon_state = "nucleardisk"
+	else
+		icon_state = "datadisk_hydro"
+
 /obj/item/disk/plantgene/attack_self(mob/user)
+	if(HAS_TRAIT(src, TRAIT_CMAGGED))
+		return
 	read_only = !read_only
 	to_chat(user, "<span class='notice'>You flip the write-protect tab to [read_only ? "protected" : "unprotected"].</span>")
 
+/obj/item/disk/plantgene/cmag_act(mob/user)
+	if(!HAS_TRAIT(src, TRAIT_CMAGGED))
+		to_chat(user, "<span class='warning'>The bananium ooze flips a couple bits on the plant disk's display, making it look just like the..!</span>")
+		ADD_TRAIT(src, TRAIT_CMAGGED, "clown_emag")
+		update_name()
+		update_desc()
+		update_icon()
+
+/obj/item/disk/plantgene/uncmag()
+	update_name()
+	update_desc()
+	update_icon()
```

Comment:
These should be converted to `update_appearance(UPDATE_NAME | UPDATE_DESC | UPDATE_ICON_STATE)` to be consistent with #17943